ydb/core/blobstorage/ut_blobstorage/deadlines.cpp

```diff
@@ -102,95 +110,426 @@ Y_UNIT_TEST_SUITE(Deadlines) {
+                        indexOnly));
+            });
+        };
+
+        TestImpl<TEvBlobStorage::TEvVGetResult, TEvBlobStorage::TEvRangeResult>(erasure, delay, timeout, send);
+    }
+
+    void TestCollectGarbage(const TBlobStorageGroupType& erasure, bool multicollect, bool hard) {
+        TDuration delay = TDuration::Seconds(50);
+        TDuration timeout = TDuration::Seconds(40);
+
+        std::unique_ptr<TVector<TLogoBlobID>> keep(new TVector<TLogoBlobID>);
+
+        auto send = [&](TestCtx& ctx) {
+            ui32 tabletId = 100;
+            ui32 generation = 10;
+            ui32 channel = 1;
+            ui32 steps = 10'000;
+            for (ui32 step = 1; step < steps; ++step) {
+                TString data = MakeData(1000);
+                TLogoBlobID blobId(tabletId, generation, step, channel, data.size(), 123);
+                if (step % 2 == 0) {
+                    keep->push_back(blobId);
+                }
+                ctx.Env->Runtime->WrapInActorContext(ctx.Edge, [&] {
+                    SendToBSProxy(ctx.Edge, ctx.GroupId, new TEvBlobStorage::TEvPut(blobId, data, TInstant::Max(),
+                            NKikimrBlobStorage::TabletLog));
+                });
+                auto putRes = ctx.Env->WaitForEdgeActorEvent<TEvBlobStorage::TEvPutResult>(ctx.Edge, false, TInstant::Max());
+                UNIT_ASSERT(putRes);
+                UNIT_ASSERT_VALUES_EQUAL(putRes->Get()->Status, NKikimrProto::OK);
+            }
+
+            TInstant now = TAppData::TimeProvider->Now();
+            ctx.Env->Runtime->WrapInActorContext(ctx.Edge, [&] {
+                SendToBSProxy(ctx.Edge, ctx.GroupId, new TEvBlobStorage::TEvCollectGarbage(tabletId, generation, 1, channel, 
+                        true, generation, steps / 2, keep.release(), nullptr, now + timeout, multicollect, hard));
+            });
+        };
+
+        TestImpl<TEvBlobStorage::TEvVCollectGarbageResult, TEvBlobStorage::TEvCollectGarbageResult>(erasure, delay, timeout, send);
     }
 
-    #define TEST_DEADLINE(method, erasure)                      \
-    Y_UNIT_TEST(Test##method##erasure) {                        \
-        Test##method(TBlobStorageGroupType::Erasure##erasure,   \
-            TDuration::Seconds(50), TDuration::Seconds(40));    \
+    void TestStatus(const TBlobStorageGroupType& erasure) {
+        TDuration delay = TDuration::Seconds(50);
+        TDuration timeout = TDuration::Seconds(40);
+
+        auto send = [&](TestCtx& ctx) {
+            TInstant now = TAppData::TimeProvider->Now();
+            ctx.Env->Runtime->WrapInActorContext(ctx.Edge, [&] {
+                SendToBSProxy(ctx.Edge, ctx.GroupId, new TEvBlobStorage::TEvStatus(now + timeout));
+            });
+        };
+
+        TestImpl<TEvBlobStorage::TEvVStatusResult, TEvBlobStorage::TEvStatusResult>(erasure, delay, timeout, send);
+    }
+
+    template<class Tag>
+    void Test(const TBlobStorageGroupType&) {}
+
+    struct TTagPut {};
```

Comment:
What are these Tags supposed to be used for?

You can create functions with the same naming system as well and get rid of the third(!) request type list of in this file